mig/shared/base.py

```diff
@@ -505,23 +523,28 @@ def force_utf8(val, highlight=''):
         val = "%s" % val
     if not is_unicode(val):
         return val
-    return "%s%s%s" % (highlight, val.encode("utf8"), highlight)
+    return codecs.encode("%s%s%s" % (highlight, val, highlight), 'utf8')
+
+
+def _walk_and_convert_recursive(input_obj, highlight='', _is_primitive=None, _force_primitive=None, _force_recursive=None):
```

Comment:
I think the last three optional args (`bla=None`) should really either be changed to mandatory or we should test that they aren't `None` before uncritically using them as functions below.

Reply:
https://github.com/ucphhpc/migrid-sync/pull/93/commits/a34927d01fd3a60cc165b48e44909f56b28d4362